torchrl/modules/tensordict_module/sequence.py

```diff
@@ -191,9 +195,57 @@ def _split_param(
             out.append(param_list[a:b])
         return out
 
+    def select_subsequence(
+        self, in_keys: Iterable[str] = None, out_keys: Iterable[str] = None
+    ) -> "TensorDictSequence":
+        """
+        Returns a new TensorDictSequence with only the modules that are necessary to compute
+        the given output keys with the given input keys.
+
+        Args:
+            in_keys: input keys of the subsequence we want to select
+            out_keys: output keys of the subsequence we want to select
+
+        Returns:
+            A new TensorDictSequence with only the modules that are necessary acording to the given input and output keys.
+        """
+        if in_keys is None:
+            in_keys = deepcopy(self.in_keys)
+        if out_keys is None:
+            out_keys = deepcopy(self.out_keys)
+        id_to_keep = set([i for i in range(len(self.module))])
+        for i, module in enumerate(self.module):
+            if all(key in in_keys for key in module.in_keys):
```

Comment:
What's the usage of selecting in_keys? I can understand why we want to restrict the outputs, but I don't really see when we want to restrict inputs.

Reply:
Also what happens if you say you want some out_keys but they conflict with the in_keys? Is the sequence going to be empty?

Reply:
We want to be able to select the in_keys to be able to directly input an intermediarry block. For example imagine you have a hidden layer that you want to inject from a precomputed tensordict, this allows to do so.

Reply:
Yes it would be empty if your out keys are before the in_keys

Reply:
Got it
We're doing something similar in #352: you can input an incomplete tensordict and only the relevant ops will be executed. I wonder if we need both ways of doing the same thing. The advantage of your implementation is that it is self-consistent though.